reinforcement_learning/common/sagemaker_rl/ray_launcher.py

```diff
@@ -231,6 +233,50 @@ def save_checkpoint_and_serving_model(self, algorithm=None, env_string=None, con
         # To ensure SageMaker local mode works fine
         change_permissions_recursive(INTERMEDIATE_DIR, 0o777)
         change_permissions_recursive(MODEL_OUTPUT_DIR, 0o777)
+        
+    def set_up_checkpoint(self, config=None):
+        try:
+            checkpoint_dir = config['training']['restore']
+            print("Found checkpoint dir %s in user config." %checkpoint_dir)
+            return config
+        except KeyError:
+            pass
+
+        try:
+            # validate files in checkpoint channel and set restore dir in ray config
+            validation = 0
+            num_checkpoint_files = len(os.listdir(CHECKPOINT_DIR))
+                
+            if num_checkpoint_files is not (2 or 3):
+                raise RuntimeError("Unexpected files in checkpoint dir.",
+                                    "Please check ray documents for the correct checkpoint format.")
+
+            checkpoint_file_in_container = ""
```

Comment:
Customer has full control on what to upload to 'checkpoint' channel (e.g. one .tune_metadata file with two random files can bypass these validation). It will be good to have more thorough validations.

Reply:
There is a tradeoff in terms of validation -- how much would SageMaker RL handle, and how much left for Ray. In case where customer points 'checkpoint' channel to a ray model checkpoint on s3, Ray takes care of the validation e2e. It's absolutely possible that customer put some random files there, and there's the reason why the number of files is checked by SageMaker. I think after that level of validation, the rest can be handled to Ray. Error message such as `cannot find .tune_metadata` should be clear to users.